exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -285,6 +285,40 @@ func TestExporterLogs(t *testing.T) {
 		rec.WaitItems(1)
 	})
 
+	t.Run("publish otel mapping mode", func(t *testing.T) {
+		rec := newBulkRecorder()
+		server := newESTestServer(t, func(docs []itemRequest) ([]itemResponse, error) {
+			rec.Record(docs)
+			return itemsAllOK(docs)
+		})
+
+		exporter := newTestLogsExporter(t, server.URL, func(cfg *Config) {
+			cfg.LogsDynamicIndex.Enabled = true
+			cfg.Mapping.Mode = "otel"
+		})
+		mustSendLogs(t, exporter, newLogsWithAttributeAndResourceMap(
+			map[string]string{
+				"data_stream.dataset": "attr.dataset",
+				"attr.foo":            "attr.foo.value",
+			},
+			map[string]string{
+				"data_stream.dataset":   "resource.attribute.dataset",
+				"data_stream.namespace": "resource.attribute.namespace",
```

Comment:
Just as a question: have we implemented sanitization for these already? For example, would `resource-attribute-dataset` be sanitized to `resource_attribute_dataset`?

Reply:
Nice catch, we have not. Will address in a different PR.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34285